File: snippets/Snippet-003.kt
package fr.o80.kata

import java.io.PrintWriter

fun main() {
    val écrivain = PrintWriter("customer-01")
    écrire(écrivain, "firstname", "Olivier")
    écrire(écrivain, "lastname", "PEREZ")
    écrivain.close()
}

fun écrire(output: PrintWriter, key: String, value: Any) {
    output.write("$key(${value.javaClass})=$value")
}

File: snippets/js/Snippet-003.js

import { PrintWriter } from './lib/PrintWriter'

function main() {
    let écrivain = new PrintWriter("customer-01")
    écrire(écrivain, "firstname", "Olivier")
    écrire(écrivain, "lastname", "PEREZ")
    écrivain.close()
}

function écrire(output, key, value) {
    output.write(`${key}(${typeof value})=${value}`)
}